docs/DeveloperGuide.adoc

```diff
@@ -452,27 +452,177 @@ with current implementations (able to wrap `ObservableList` and sort the content
 ** Cons: Cannot make use of existing methods in ORGANice that work for Comparator.
 // end::sort[]
 
+// tag::processingDG[]
+
 === Processing feature
 This section describes how the processing feature is implemented.
 
-Processing is done by accessing `Patient` and `Donor` in `ModelManager`, then creating an ArrayList named, `ProcessingList` to store the tasks needed to do.
+Processing is facilitated by the `ProcessingCommand` class, which extends the `Command` class.
+
+Processing is done by accessing `Patient` and `Donor` in `ModelManager`,
+then creating a `ProcessingList` which acts like ArrayList to store the tasks which needs to be done.
+
+It implements the `ProcessingCommand#isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model)` boolean operation. +
+This method will first determine which Nric belongs to a patient and which Nric belongs to a donor
+since the input allows user to give two Nric in no particular order. +
+It will also get the `Donor` and `Patient` objects with the NRIC given. +
+Then, it will check if the donor have a `ProcessingList` generated before. +
+If the donor have an empty `ProcessingList`, it means that the donor is newly processed. +
+Hence, it will mean that the donor may be able to pair with the given patient. +
+If the donor already have a `ProcessingList`, it means that the donor is already being processed with a patient. +
+Then, this method will check if the patient given by the user is the same patient being processed together with this donor. +
+This is done by checking the `Patient` attribute that is tagged to the `ProcessingList` of the donor. +
+Finally, the method will check if the patient and donor given by the user is valid by checking if
+they are able to be matched or not and if they exist in ORGANice. +
+A code snippet of the method is shown below: This code aims to get the `Donor` and `Patient` from the given Nrics given by the user. +
+This method is necessary since users able to give the Nrics input in no particular order.
+```
+if (model.hasDonor(firstNric)) {
+    donorNric = firstNric;
+    donor = model.getDonor(donorNric);
+
+    patientNric = secondNric;
+    patient = model.getPatient(patientNric);
+} else {
+    patientNric = firstNric;
+    patient = model.getPatient(patientNric);
+
+    donorNric = secondNric;
+    donor = model.getDonor(donorNric);
+}
+```
+
+`Status` class is also  created and it is one of the attributes of `Patient` and `Donor`.
+However, this cannot be edited or added by users. +
+Processing command make changes to the `Status` of the patient and donor.
```

Comment:
Is this a better way of phrasing: The status of processing for the 'Patient' and 'Donor' will change when a 'Processing' command is executed on that pair of patient and donor.

Reply:
amended